### PR TITLE
Get cudf/spark dependency from the correct .m2 dir

### PR DESCRIPTION
**Pull request by NvTimLiu**
Bug: https://github.com/NVIDIA/spark-rapids/issues/1031

'WORKSPACE' & 'M2DIR' vars are needed for shims to gen the correct cudf/spark dependency info in shims.

Below error in 'spark*-info.properties' is due to unset of 'WORKSPACE' & 'M2DIR':
    build/dependency-info.sh: line 30: /jenkins/printJarVersion.sh: No such file or directory
    build/dependency-info.sh: line 33: /jenkins/printJarVersion.sh: No such file or directory

To fix the error, we set the default values for them in 'build/dependency-info.sh':
    'M2DIR=$HOME/.m2/repository'
    'WORKSPACE=../..'

We also need to explicitly set the correct 'M2DIR' path, in case we change it by '-Dmaven.repo.local=$M2DIR'.
Already updated Jenkins scripts to set the correct 'M2DIR'.

Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

**Review by revans2 — DISMISSED** [review]

**Comment by jlowe** [comment]
The changes look good, but it doesn't completely fix the issue.  If the script fails to run successfully for some reason then it will still not fail the build.  I would like to see changes related to error handling in addition to fixing the paths when run outside of Jenkins.  The improved error handling doesn't have to be part of this PR, but we shouldn't close #1031 until that is addressed.

**Comment by NvTimLiu** [comment]
@jlowe @revans2,  I agree if the script fails again later, then we'll stop the build.

I tried to add some scripts to check how to let `mvn package`  fails in case the script 'build/dependency-info.sh' runs failure.

As the `dependency-info.sh` is script is running in the child process forked by `mvn package`,  but the mvn process does not check the forked process's status, so even the child process fails, `mvn` process still run builds.

I did not find a good way to let `mvn` process automatically stop when 'dependency-info.sh' fails.

Have you any suggestions?

mvn process:                      `21245`   18897        java   -Dmaven.home=/home/timl/anaconda3/opt/maven
dependency-info.sh:           `21304`   `21245`     bash  build/dependency-info.sh 0.17-SNAPSHOT cuda10-1 3.0.0

**Comment by jlowe** [comment]
> I did not find a good way to let mvn process automatically stop when 'dependency-info.sh' fails.

The subprocess is launched by the Maven antrun plugin which in turn is using Ant to perform the tasks.  In this case it's the Ant `exec` task which is documented [here](https://ant.apache.org/manual/Tasks/exec.html).  There is a `failonerror` attribute that defaults to `false`.  Specifying that attribute as `true` will cause the build to fail if the program exits with an error code.  This should be done for every place in the build using the Ant `exec` task to ensure the build fails if a program we're trying to launch as part of the build fails.

That's the first step.  The next step is to ensure the script will fail if anything launched by the script fails.  That can be done by placing a `set -e` or `set -o errexit` command as the first command in the script.  Alternatively you can add a `-e` argument to every `bash` command line in an antrun `exec` directive.

**Comment by NvTimLiu** [comment]
@jlowe I tried the way you suggested, the `failonerror` attribute works for me.
Thanks very much for your help. I'll update it on the PR.

File: build/dependency-info.sh [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #   SPARK_VER - The version of spark
 
 # Parse cudf and spark dependency versions
+set -e
```
Comment (jlowe) [inline build/dependency-info.sh:26]: The same needs to happen in `build/build-info` and the corresponding `failonerror` pom change is needed in the top-level pom's use of the antrun plugin.

**Comment by NvTimLiu** [comment]
build

File: build/dependency-info.sh (hunk 1 above)
Comment (NvTimLiu) [reply build/dependency-info.sh:26]: > The same needs to happen in `build/build-info` and the corresponding `failonerror` pom change is needed in the top-level pom's use of the antrun plugin.

@jlowe Thanks very much for reminding me, updated.

Comment (jlowe) [reply build/dependency-info.sh:26]: Needs 2020 copyrights

**Comment by NvTimLiu** [comment]
build

**Review by revans2 — DISMISSED** [review]

File: build/build-info [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 #
-# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
```
Comment (jlowe) [inline build/build-info:4]: This needs to be 2019-2020.  We don't give up copyright for older years.

**Comment by jlowe** [comment]
build

**Comment by NvTimLiu** [comment]
build

File: build/build-info (hunk 2 above)
Comment (NvTimLiu) [reply build/build-info]: > This needs to be 2019-2020. We don't give up copyright for older years.

Ok

**Comment by jlowe** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**